client/app/scripts/charts/edge-container.js

```diff
@@ -71,24 +72,29 @@ class EdgeContainer extends React.Component {
   }
 
   preparePoints(nextPoints) {
-    // Spring needs constant field count, hoping that dagre will insert never more than `pointCount`
-    let { pointsMap } = this.state;
-
-    // filling up the map with copies of the first point
-    const filler = nextPoints.first();
-    const missing = pointCount - nextPoints.size;
-    let index = 0;
-    if (missing > 0) {
-      while (index < missing) {
-        pointsMap = pointsMap.set(`x${index}`, spring(filler.get('x'), animConfig));
-        pointsMap = pointsMap.set(`y${index}`, spring(filler.get('y'), animConfig));
-        index++;
-      }
+    nextPoints = nextPoints.toJS();
+
+    // Motion requires a constant number of waypoints along the path of each edge
+    // for the animation to work correctly, but dagre might be changing their number
+    // depending on the dynamic topology reconfiguration. Here we are transforming
+    // the waypoints array given by dagre to the fixed size of `WAYPOINTS_CAP` that
+    // Motion could take over.
+    const pointsMissing = WAYPOINTS_CAP - nextPoints.length;
+    if (pointsMissing > 0) {
+      // Whenever there are some waypoints missing, we simply populate the beginning of the
+      // array with the first element, as this leaves the curve interpolation unchanged.
+      nextPoints = Array(pointsMissing).fill(_.first(nextPoints)).concat(nextPoints);
```

Comment:
Not sure, but I think it'd be better to replace it with `_.times(pointsMissing, nextPoints[0])` anyway.